### PR TITLE
Issue 846, 840 & More

### PR DESCRIPTION
**Pull request by tamimfatahi**
- Added `dynesty` back in for Window users only. In the future, we'll detect a c++ compiler and allow users to install `ultranest` from the `requirements.txt`. 
- Sorted phase calculations to one function in `elca.py`.
- Moved most plots outside of exotic.py into their own file called `plots.py`. May move this later on to `api/plotting.py`.
- Added the ability for users to input comparison star used during photometry for pre-reduced data reduction. Also updated these changes in the GUI.
- Added a few filters to `filters.py`.
- Other code clean-ups.
- Reorganized paths on how we save files/pngs/pdfs.

I'll tag you guys on further things I want to make sure are aware of before pulling to `develop`.

Edit: Also, all future centroids are going on one pdf/png shown here:
CentroidPositions Distances_HAT-P-32 b_17-December-2017

### CODE REVIEW

**Comment by pearsonkyle** [comment]
> * Added `dynesty` back in for Window users only. In the future, we'll detect a c++ compiler and allow users to install `ultranest` from the `requirements.txt`.
> * Sorted phase calculations to one function in `elca.py`.
> * Moved most plots outside of exotic.py into their own file called `plots.py`. May move this later on to `api/plotting.py`.
> * Added the ability for users to input comparison star used during photometry for pre-reduced data reduction. Also updated these changes in the GUI.
> * Added a few filters to `filters.py`.
> * Other code clean-ups.
> * Reorganized paths on how we save files/pngs/pdfs.
> 
> I'll tag you guys on further things I want to make sure are aware of before pulling to `develop`.
> 
> Edit: Also, all future centroids are going on one pdf/png shown here:
> CentroidPositions Distances_HAT-P-32 b_17-December-2017

Please check for spelling on the graph axes. 


Also, why are the delta position measurements at integer precision? The plots above them seem to suggest we're measuring centroid positions with greater precision and thus should be capable of plots that aren't as flat looking. 


I ran these changes on 3 other light curves and it behaves properly, if anything it's faster than before. I had to add a line of code to the plotting script otherwise, it looks good to me!

**Review by tamimfatahi — COMMENTED** [review]
> Please check for spelling on the graph axes.
> 
> Also, why are the delta position measurements at integer precision? The plots above them seem to suggest we're measuring centroid positions with greater precision and thus should be capable of plots that aren't as flat looking.
> 
> I ran these changes on 3 other light curves and it behaves properly, if anything it's faster than before. I had to add a line of code to the plotting script otherwise, it looks good to me!

Noticed that error - will change it right now. 

I'll change the centroid distance plots from int to floats as well.

Thanks for your help!

**Comment by tamimfatahi** [comment]
New plots for Centroids:

CentroidPositions Distances_HAT-P-32 b_17-December-2017

File: exotic/api/elca.py [hunk 1]
```diff
@@ -368,21 +376,77 @@ def prior_transform(upars):
...
+            chis = []
+            for i in range(len(tests)):
+                lightcurve = transit(self.time, tests[i])
+                tests[i]['a1'] = mc_a1(tests[i]['a2'], 0, lightcurve, self.airmass, self.data)[0]
+                airmass = tests[i]['a1'] * np.exp(tests[i]['a2'] * self.airmass)
+                residuals = self.data - (lightcurve * airmass)
+                chis.append(np.sum(residuals ** 2))
+
+            mi = np.argmin(chis)
+            self.parameters = copy.deepcopy(tests[mi])
```
[the remark this answers is not in the document]
Comment (pearsonkyle) [reply exotic/api/elca.py:449]: Yeah, as much as I like ultranest we should support all OS's so dynesty is fine.

**Review by pearsonkyle — APPROVED** [review]
Looks good to me. I ran it on 3 datasets from the Kuiper 61" and it passed

**Pull request merged**